signale.js

```diff
@@ -62,7 +62,7 @@ class Signale {
     const {stack} = new Error();
     Error.prepareStackTrace = _;
 
-    const callers = stack.map(x => path.basename(x.getFileName()));
+    const callers = stack.map(x => x.getFileName() && path.basename(x.getFileName()));
```

Comment:
By replacing the lines `65`-`66` of `signale.js` with the following:

```js
// signale.js 
// ...
const callers = stack.map(x => x.getFileName()); // L65
console.log(callers); // L66
// ...
```

and then running example of #5 on both node `8.9.1` and `9.11.1`:

```js
// file path: home/signale/foo.js
const signale = require('signale');
signale.config({displayFilename: true});

Promise.resolve().then(() => {
  signale.error('Foo');
});
```
we get these file paths, returned by the callsites:

```js
// Filepaths returned by callsites
[ 'home/signale/foo.js',
  'home/signale/foo.js',
  'home/signale/foo.js',
  'home/signale/foo.js',
  'home/signale/foo.js',
  'home/signale/foo.js',
  null,
  'internal/process/next_tick.js',
  'module.js',
  'bootstrap_node.js' ]
```

Thus, the issue is indeed `path.basename(x.getFileName())`, that crushes once `null`, returned by the `<anonymous>` callsite, is pass to it as an argument. So, instead of returning `unknown`, we can still return the filename `foo.js` just by calling `path.basename` at the end of `filename()`, e.g:

```js
  get filename() {
    const _ = Error.prepareStackTrace;
    Error.prepareStackTrace = (error, stack) => stack;
    const {stack} = new Error();
    Error.prepareStackTrace = _;

    const callers = stack.map(x => x.getFileName()); // Remove path.basename()

    const filePath = callers.find(x => {  // Get the whole filepath
      return x !== callers[0];
    });

    return path.basename(filePath);  // Return the basename
  }
```

It is initial idea, so please feel free to share your thoughts : )

Reply:
I completely agree, we can then change the return statement to something like this: 

```js
return firstExternalFilePath ? path.basename(firstExternalFilePath) : 'unknown';
```

This way even the most extreme cases can be covered and also avoid any potential errors on the way